lib/provisioning/Provisioner.js

```diff
@@ -46,36 +72,39 @@ var parameterValidation = {
         {regex : /^[a-z\.0-9:-]+$/, example : 'example.com or localhost'},
 };
 
-Provisioner.prototype.validate = function(parameters) {
-    let parameterNames = ['matrix_room_id', 'remote_room_channel', 'remote_room_server'];
-    for (var i = 0; i < parameterNames.length; i++) {
-        let name = parameterNames[i];
-
-        let actual = parameters[name];
-        let valid = parameterValidation[name];
+Provisioner.prototype._validate = function(actual, parameterName) {
+    let valid = parameterValidation[parameterName];
+    if (!actual) {
+        throw new Error(
+            `${parameterName} not provided (like '${valid.example}').`
```

Comment:
Careful! What if `parameterName` is not one of the three you expect? This will `TypeError` rather than give you this error because of `valid.example`.